src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -1646,18 +1645,38 @@ def aks_show(cmd, client, resource_group_name, name):
 
 def aks_update_credentials(cmd, client, resource_group_name, name,
                            reset_service_principal=False,
+                           reset_aad=False,
                            service_principal=None,
                            client_secret=None,
+                           aad_server_app_id=None,
+                           aad_server_app_secret=None,
+                           aad_client_app_id=None,
+                           aad_tenant_id=None,
                            no_wait=False):
-    if reset_service_principal != 1:
-        raise CLIError('Please specify "--reset-service-principal".')
-    if service_principal is None or client_secret is None:
-        raise CLIError('Please specify --service-principal and --client-secret '
-                       'when --reset-service-principal flag is on.')
+    if reset_service_principal + reset_aad != 1:
+        raise CLIError('Please specify one of "--reset-service-principal" or "--reset-aad-profile".')
```

Comment:
Usage statement syntax preferred:
`usage error: --reset-service-principal | --reset-aad-profile`

Reply:
For the usage error, the code base appears to have a mixture of capitalizing vs lowercase`Usage Error` vs. `usage error`. Is one preferred over the other?

Reply:
lower case because that it what argparse uses and we essentially emulate them.

Reply:
Ok, thanks